arquillian-environment-reporter/arquillian-environment-reporter-impl/src/main/java/org/arquillian/environment/reporter/impl/ArquillianEnvironmentReporterLifecycleManager.java

```diff
@@ -0,0 +1,64 @@
+package org.arquillian.environment.reporter.impl;
+
+import org.arquillian.reporter.api.builder.Reporter;
+import org.arquillian.reporter.api.event.SectionEvent;
+import org.arquillian.reporter.api.event.TestSuiteConfigurationSection;
+import org.arquillian.reporter.api.model.report.ConfigurationReport;
+import org.jboss.arquillian.core.api.Event;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.event.suite.BeforeSuite;
+
+import java.nio.charset.Charset;
+import java.util.TimeZone;
+
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.CHARSET;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.DOCKER;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.ENVIRONMENT_SECTION_NAME;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.JAVA_VERSION;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.OPERATIVE_SYSTEM;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.TEST_RUNNER;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.TIMEZONE;
+
+public class ArquillianEnvironmentReporterLifecycleManager {
```

Comment:
`ArquillianEnvironmentReporterLifecycleManager` - it's not really what this class does. It hooks to the event mechanism and create report. So there is no management of lifecycle going on for my taste :)

Isn't `ArquillianEnvironmentReportCreator`  reflecting better the responsibility of this class?